### PR TITLE
udev-block-notify: init at 0.7.11

### PR DESCRIPTION
**Pull request by danbulant**
## Description of changes

Adds [udev-block-notify](https://github.com/eworm-de/udev-block-notify), a user daemon that sends a notification about block device changes.

I am new to NixOS and Nix, just porting software I used to use before.  
Also added myself as a maintainer for that package.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md). Hopefully?

As a sort of closing question, does it make sense to try and make it a service here in nixpkgs, in home-manager or should I not do that at all? The current package creates a user systemd service that will simply start the program in background, but doesn't enable it.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    sha256 = "sha256-A0uhfb2mEAAJgxRkv+MWTk/9oFiz3r7deAlu1Kpk+CI=";
+  };
+
+  nativeBuildInputs = [ multimarkdown coreutils pkg-config ];
+  buildInputs = [ libnotify udev systemdLibs glib ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/share/doc/udev-block-notify/screenshots
+    mkdir -p $out/lib/systemd/user
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:30]:
```suggestion
```

`install` will create parent directory

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    install -D -m0644 screenshots/usb.png $out/share/doc/udev-block-notify/screenshots/usb.png
+    install -D -m0644 screenshots/optical.png $out/share/doc/udev-block-notify/screenshots/optical.png
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/eworm-de/udev-block-notify";
+    description = "Notify about udev block events";
+    mainProgram = "udev-block-notify";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ danbulant ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:45]:
```suggestion
    maintainers = with maintainers; [ danbulant ];
    platforms = lib.platforms.linux
```

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    install -D -m0755 udev-block-notify $out/bin/udev-block-notify
+    install -D -m0644 systemd/udev-block-notify.service $out/lib/systemd/user/udev-block-notify.service
+
+    install -D -m0644 README.md $out/share/doc/udev-block-notify/README.md
+    install -D -m0644 README.html $out/share/doc/udev-block-notify/README.html
+    install -D -m0644 screenshots/usb.png $out/share/doc/udev-block-notify/screenshots/usb.png
+    install -D -m0644 screenshots/optical.png $out/share/doc/udev-block-notify/screenshots/optical.png
+  '';
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:40]: Better to avoid `with lib`

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    mkdir -p $out/bin
+    mkdir -p $out/share/doc/udev-block-notify/screenshots
+    mkdir -p $out/lib/systemd/user
+    install -D -m0755 udev-block-notify $out/bin/udev-block-notify
+    install -D -m0644 systemd/udev-block-notify.service $out/lib/systemd/user/udev-block-notify.service
+
+    install -D -m0644 README.md $out/share/doc/udev-block-notify/README.md
+    install -D -m0644 README.html $out/share/doc/udev-block-notify/README.html
+    install -D -m0644 screenshots/usb.png $out/share/doc/udev-block-notify/screenshots/usb.png
+    install -D -m0644 screenshots/optical.png $out/share/doc/udev-block-notify/screenshots/optical.png
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:37]:
```suggestion
    runHook preInstall

    install -D -m0755 udev-block-notify $out/bin/udev-block-notify
    install -D -m0644 systemd/udev-block-notify.service $out/lib/systemd/user/udev-block-notify.service

    install -D -m0644 README.md $out/share/doc/udev-block-notify/README.md
    install -D -m0644 README.html $out/share/doc/udev-block-notify/README.html
    install -D -m0644 screenshots/usb.png $out/share/doc/udev-block-notify/screenshots/usb.png
    install -D -m0644 screenshots/optical.png $out/share/doc/udev-block-notify/screenshots/optical.png

    runHook postInstall
```

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+}:
+stdenv.mkDerivation {
+  pname = "udev-block-notify";
+  version = "0.7.11";
+
+  src = fetchFromGitHub {
+    owner = "eworm-de";
+    repo = "udev-block-notify";
+    rev = "e652f54edd9ea87c784d796435967a18189d1936";
+    sha256 = "sha256-A0uhfb2mEAAJgxRkv+MWTk/9oFiz3r7deAlu1Kpk+CI=";
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:21]:
```suggestion
    hash = "sha256-A0uhfb2mEAAJgxRkv+MWTk/9oFiz3r7deAlu1Kpk+CI=";
```

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  lib
+}:
+stdenv.mkDerivation {
+  pname = "udev-block-notify";
+  version = "0.7.11";
+
+  src = fetchFromGitHub {
+    owner = "eworm-de";
+    repo = "udev-block-notify";
+    rev = "e652f54edd9ea87c784d796435967a18189d1936";
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:20]:
```suggestion
    rev = "refs/tags/${finalAttrs.version}";
```

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  multimarkdown,
+  coreutils,
+  libnotify,
+  udev,
+  systemdLibs,
+  glib,
+  pkg-config,
+  lib
+}:
+stdenv.mkDerivation {
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:13]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/eworm-de/udev-block-notify";
+    description = "Notify about udev block events";
+    mainProgram = "udev-block-notify";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ danbulant ];
+  };
+}
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:47]:
```suggestion
})
```

**Review by Aleksanaa — COMMENTED** [review]
Please format package.nix with nixfmt-rfc-style, and squash into only two commits:

```
maintainers: add danbulant
udev-block-notify: init at 0.7.11
```

**Comment by danbulant** [comment]
Thanks a lot for your guidance!

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  src = fetchFromGitHub {
+    owner = "eworm-de";
+    repo = "udev-block-notify";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-A0uhfb2mEAAJgxRkv+MWTk/9oFiz3r7deAlu1Kpk+CI=";
+  };
+
+  nativeBuildInputs = [
+    multimarkdown
+    coreutils
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:26]:
```suggestion
```

Not needed

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 10]
```diff
@@ -0,0 +1,58 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  multimarkdown,
+  coreutils,
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:5]:
```suggestion
```

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+  src = fetchFromGitHub {
+    owner = "eworm-de";
+    repo = "udev-block-notify";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-A0uhfb2mEAAJgxRkv+MWTk/9oFiz3r7deAlu1Kpk+CI=";
+  };
+
+  nativeBuildInputs = [
+    multimarkdown
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:25]: Is this really used?

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+    udev
+    systemdLibs
+    glib
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D -m0755 udev-block-notify $out/bin/udev-block-notify
+    install -D -m0644 systemd/udev-block-notify.service $out/lib/systemd/user/udev-block-notify.service
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:40]:
```
postPatch = ''
  substituteInPlace systemd/udev-block-notify.service \
    --replace-fail '/usr/bin/udev-block-notify' "$out/bin/udev-block-notify"
'';
```

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+    install -D -m0644 screenshots/optical.png $out/share/doc/udev-block-notify/screenshots/optical.png
+
+    runHook postInstall
+  '';
+
+  meta = {
+    homepage = "https://github.com/eworm-de/udev-block-notify";
+    description = "Notify about udev block events";
+    mainProgram = "udev-block-notify";
+    license = lib.licenses.gpl3Only;
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:54]:
```suggestion
    license = lib.licenses.gpl3Plus;
```

https://github.com/eworm-de/udev-block-notify/tree/main?tab=readme-ov-file#license-and-warranty

File: pkgs/by-name/ud/udev-block-notify/package.nix (hunk 11 above)
Comment (danbulant) [reply pkgs/by-name/ud/udev-block-notify/package.nix:25]: Makefile uses this during build to generate the HTML documentation.
I could disable that if that's not desired.

File: pkgs/by-name/ud/udev-block-notify/package.nix (hunk 9 above)
Comment (danbulant) [reply pkgs/by-name/ud/udev-block-notify/package.nix:26]: I wasn't sure if install was always available without it, thanks

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+    glib
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D -m0755 udev-block-notify $out/bin/udev-block-notify
+    install -D -m0644 systemd/udev-block-notify.service $out/lib/systemd/user/udev-block-notify.service
+
+    substituteInPlace systemd/udev-block-notify.service \
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:40]: Wait, you first install it, then replace the source file???

File: pkgs/by-name/ud/udev-block-notify/package.nix [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+      --replace-fail '/usr/bin/udev-block-notify' "$out/bin/udev-block-notify"
+
+    install -D -m0644 README.md $out/share/doc/udev-block-notify/README.md
+    install -D -m0644 README.html $out/share/doc/udev-block-notify/README.html
+    install -D -m0644 screenshots/usb.png $out/share/doc/udev-block-notify/screenshots/usb.png
+    install -D -m0644 screenshots/optical.png $out/share/doc/udev-block-notify/screenshots/optical.png
+
```
Comment (Aleksanaa) [inline pkgs/by-name/ud/udev-block-notify/package.nix:47]:
```suggestion
```
We usually don't add those, unless they are used by the main executable

File: pkgs/by-name/ud/udev-block-notify/package.nix (hunk 14 above)
Comment (danbulant) [reply pkgs/by-name/ud/udev-block-notify/package.nix:37]: I missed that, sorry

**Review by FliegendeWurst — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**